### DISCUSSION TITLE
Nasty errors in the console when on a non ENS-supported chain

### DISCUSSION THREAD

**Discussion by holma91** [rainbow-me/rainbowkit#429] [Q&A] [answered]
```
Error: network does not support ENS
(operation="lookupAddress", network="matic", code=UNSUPPORTED_OPERATION, version=providers/5.6.6)
```
and 
```
Error: network does not support ENS
(operation="getAvatar", network="matic", code=UNSUPPORTED_OPERATION, version=providers/5.6.6)
```

So to me it looks like the functions lookupAddress and getAvatar gets called when they shouldn't. How do I prevent them from running when I'm on a non-ethereum blockchain such as polygon?

**Comment by holma91** [comment]
Looks like this has something to do with development mode with next.js. Errors are gone if I run in production mode.

**Reply by lucasfernandes** [reply]
Yeah, tested here and the problem doesn't happen on production mode. Thanks for the tip.

**Reply by songkeys** [reply]
The production mode just suppresses the error message, not eliminates the error itself. I think the library should provide a config to disable ens requests on demand.

**Comment by markdalgleish** [comment, marked as answer]
As of [RainbowKit v0.4.1,](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%400.4.1) we only ever request ENS names and avatars from mainnet, and in [RainbowKit v0.4.4](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%400.4.4) we now only make ENS requests when mainnet is in your list of configured chains which will prevent these errors from occurring.
